### ISSUE TITLE
Problem with TimePicker in Dialogbox

### ISSUE THREAD

**Issue by HansRom** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#410]
We are trying to use a timepicker in a dialogbox (The same as in sample 3 from your demo), but when we open the Timepicker the time is always empty. We found out that in the code: 

```
private void SetSelectedTime(bool beCautious = false)
        {
            if (!string.IsNullOrEmpty(_textBox?.Text))
            {
                ParseTime(_textBox.Text, t =>
                {
                    if (!beCautious || DateTimeToString(t) == _textBox.Text)
                        SetCurrentValue(SelectedTimeProperty, t);
                });
            }
            else
            {
                SetCurrentValue(SelectedTimeProperty, null);
            }
        }
```

_textBox is always "Nothing". When we use exactly the same code without the: "DialogHost.Show", the time is correctly filled in. And _textBox is not nothing but contains the correct time.
How can we fix this?

**Comment by JamesIrish** [comment]
Ditto suffering from something similar.  Fairly standard TimePicker within a dialog null's the value it's bound to when displaying.  Sets the value correctly though.

```xaml
<materialDesign:TimePicker Width="100" Margin="5 0" materialDesign:HintAssist.Hint="Time" SelectedTime="{Binding ParameterValue, UpdateSourceTrigger=PropertyChanged, Mode=TwoWay}"/>
```

**Comment by JamesIrish** [comment]
* NuGet package version 2.0.0.638

**Comment by Keboo** [comment]
This appears to be fixed in the 4.2.1 release
